toke.c

```diff
@@ -7555,14 +7566,7 @@ Perl_yylex(pTHX)
         return yyl_backtick(aTHX_ s);
 
     case '\\':
-	s++;
-	if (PL_lex_inwhat == OP_SUBST && PL_lex_repl == PL_linestr
-	 && isDIGIT(*s))
-	    Perl_ck_warner(aTHX_ packWARN(WARN_SYNTAX),"Can't use \\%c to mean $%c in expression",
-			   *s, *s);
-	if (PL_expect == XOPERATOR)
-	    no_op("Backslash",s);
-	OPERATOR(REFGEN);
+        return yyl_backslash(aTHX_ s + 1);
```

Comment:
should not we preserve the `s++`

Reply:
`s` is a local variable, so given that we're doing a `return` here, I don't think there's much benefit in updating the local variable before passing it to the callee